src/Collections-Unordered/Dictionary.class.st

```diff
@@ -391,6 +391,30 @@ Dictionary >> declare: key from: aDictionary [
 			[self add: key -> nil]
 ]
 
+{ #category : #undeclared }
+Dictionary >> declareVariable: newGlobal from: aDictionary [ 
+	"Add aGlobal to the receiver. If key already exists, do nothing. If aDictionary 
+	includes key, then remove it from aDictionary and use its association as 
+	the element of the receiver."
+
+	| globalName |
+	globalName :=  newGlobal key.
+	self associationAt: globalName ifPresent:  [:existingGlobal |
+		"need to take care to migrate existing variables to new global if class if different"
+		(existingGlobal class == newGlobal class) 
+			ifTrue: [^self].
+		newGlobal value: existingGlobal value.
+		self removeKey: globalName.
+		self add: newGlobal.
+		].
+	(aDictionary includesKey: globalName)
+		ifTrue:  [
+			self add: ((aDictionary associationAt: globalName) primitiveChangeClassTo: ClassVariable new).
+			aDictionary removeKey: globalName]
+		ifFalse: [
+			self add: newGlobal]
+]
+
```

Comment:
Again. I don't see why this should be in Dictionary. This is no general use of dictionary.
Either we need a special "variable" dictionary. Or all this logic should be in the client.